File: java/src/main/java/cn/az/code/other/bst/SameTree.java
package cn.az.code.other.bst;

import cn.az.code.common.TreeNode;

/**
 * @author az
 * @since 09/15/20
 */
public class SameTree {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null && q == null) {
            return true;
        }
        if (p == null || q == null || p.val != q.val) {
            return false;
        }

        return isSameTree(p.left, q.left)
                && isSameTree(p.right, q.right);
    }
}

File: archive/python/easy/SameTree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


def same(p: TreeNode, q: TreeNode):
    if p and q:
        return p.val == q.val and same(p.left, q.left) and same(p.right, q.right)
    return p is q


if __name__ == '__main__':
    p = TreeNode(1)
    q = TreeNode(1)
    print(same(p, q))
